doc/adrs/140_add_collapse_and_expand_services.adoc

```diff
@@ -0,0 +1,62 @@
+= ADR-138 Add collapse and expand services
+
+== Context
+
+Sirius Web doesn't provide a way for specifiers to define tools that collapse/expand nodes as part of other operations (e.g. to expand a collapsed node when a child node is created).
+
+We will add a new `diagramServices` variable  with services to collapse/expand nodes from AQL expressions and Java services.
+
+== Decision
+
+We will create a new interface `IDiagramServices` in `org.eclipse.sirius.components.collaborative.diagrams.api` with the following signature:
+```java
+public interface IDiagramServices {
+    void collapse(IDiagramServicesContext context, List<Node> nodes);
+    void expand(IDiagramServicesContext context, List<Node> nodes);
+}
+```
+
+We will add a new variable `diagramServices : IDiagramServiceContext` in the `VariableManager`.
+The `IDiagramServicesContext` would be a wrapper for objects that can be accessed by `IDiagramServices` services.
+For the moment it would only define a getter on the `DiagramContext`, which is required in our implementation of collapse/expand.
+
+This will allows developers to call these services from their AQL expressions as follows:
+```
+aql:diagramServices.collapse(Sequence{selectedNode})
+```
+
+And from Java services as follows:
+```java
+private IDiagramServices diagramServices;
+
+public EObject doStuff(Diagram previousDiagram, EObject eObject, IDiagramServicesContext diagramServicesContext) {
+  diagramServices.collapse(diagramServicesContext, previousDiagram.getNodes());
+  return eObject;
+}
+```
+
+The implementation of the `IDiagramServices` interface will be located in `org.eclipse.sirius.components.collaborative.diagrams`.
+
+The collapse/expand services will store instances of `UpdateCollapsingStateEvent` in a new attribute `DiagramContext#updateCollapsingStateEvents`.
```

Comment:
Just make the existing diagram events a list, no need to create one new attribute per event type.

Reply:
done